Runtime/Core/ShrinkablePolygon.cs

```diff
@@ -0,0 +1,1051 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Cinemachine.Utility;
+using ClipperLib;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Cinemachine
+{
+    /// <summary>
+    /// ShrinkablePolygon represents points with normals that can shrink down to it's skeleton.  
+    /// </summary>
+    internal class ShrinkablePolygon
+    {
+        /// <summary>
+        /// 2D point with shrink direction.
+        /// </summary>
+        internal class ShrinkablePoint2
+        {
+            public Vector2 m_position;
+            public Vector2 m_originalPosition;
+            public Vector2 m_shrinkDirection;
+            public bool m_cantIntersect;
+            internal static readonly Vector2 m_vector2NaN = new Vector2(float.NaN, float.NaN);
+
+            internal ShrinkablePoint2()
+            {
+                m_originalPosition = m_vector2NaN;
+            }
+
+            internal ShrinkablePoint2(Vector2 mPosition, Vector2 mOriginalPosition, Vector2 mShrinkDirection, 
+                bool mCantIntersect)
+            {
+                m_position = mPosition;
+                m_originalPosition = mOriginalPosition;
+                m_shrinkDirection = mShrinkDirection;
+                m_cantIntersect = mCantIntersect;
+            }
+        }
+
+        internal List<ShrinkablePoint2> m_points;
```

Comment:
I'm not going to keep going with the "public not internal" thing.  If the class as a whole is marked internal, then its non-private members should be public so that its API is clear.  It makes no sense to have redundant internals.  Please check remaining classes for this.